### PR TITLE
Ignore count during GPU accumulation

### PR DESCRIPTION
**Pull request by olaaun**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

FYI @oyving

### CODE REVIEW

**Comment by bratseth** [comment]
Shouldn't we fix the plus method in NodeResources instead?

**Comment by olaaun** [comment]
> Shouldn't we fix the plus method in NodeResources instead?

Seems like it's used for several things, was unsure if other method callers expect count > 1. It'd be great if we can just change `NodeResources::plus` instead

**Comment by bratseth** [comment]
@mpolden should review this ...

File: config-provisioning/src/main/java/com/yahoo/config/provision/NodeResources.java [hunk 1]
```diff
@@ -144,7 +144,9 @@ private boolean lessThan(GpuResources other) {
         public static GpuResources getDefault() { return zero; }
 
         public GpuResources plus(GpuResources other) {
-            return new GpuResources(this.count + other.count, this.memoryGb + other.memoryGb);
+            var thisMem = this.count() * this.memoryGb();
+            var otherMem = other.count() * other.memoryGb();
+            return new NodeResources.GpuResources(1, thisMem + otherMem);
```
Comment (mpolden) [inline config-provisioning/src/main/java/com/yahoo/config/provision/NodeResources.java:149]: Shouldn't `minus` have the same change?

**Review by mpolden — DISMISSED** [review]

**Comment by bratseth** [comment]
We're missing some unit tests here.

**Comment by olaaun** [comment]
Added test in https://github.com/vespa-engine/vespa/pull/28587

**Pull request merged**